src/App/app.jsx

```diff
@@ -28,25 +29,28 @@ const App = ({ root_store }) => {
             <MobxProvider store={root_store}>
                 {
                     root_store.ui.is_mobile || (root_store.ui.is_tablet && isTouchDevice) ?
-                        <Wip /> :
+                        <React.Suspense fallback={<UILoader />}>
+                            <Wip />
+                        </React.Suspense> :
                         <React.Fragment>
                             <Header />
+                            <React.Suspense fallback={<UILoader />} >
+                                <PushNotification />
+                            </React.Suspense>
                             <ErrorBoundary>
                                 <AppContents>
                                     <Routes />
-                                    <PushNotification />
                                 </AppContents>
-                                <UnsupportedContractModal />
-                                <DenialOfServiceModal />
-                                <MarketUnavailableModal />
-                                <ServicesErrorModal />
+                                <React.Suspense fallback={<UILoader />}>
```

Comment:
@easteregg I'd be better to move the suspense inside of the `Modal` component.

Reply:
Since `Modals` itself is lazy-loaded, I think it is better to have it here.